lib/puppet/pops/lookup/lookup_adapter.rb

```diff
@@ -417,11 +417,11 @@ def initialize_env_provider(lookup_invocation)
       elsif ep_config.version >= 5
         unless provider_name.nil? || Puppet[:strict] == :off
           Puppet.warn_once(:deprecation, 'environment.conf#data_provider',
-            "Defining environment_data_provider='#{provider_name}' in environment.conf is deprecated", env_path + 'environment.conf')
+            _("Defining environment_data_provider='%{provider_name}' in environment.conf is deprecated") % { provider_name: provider_name }, env_path + 'environment.conf')
```

Comment:
@Magisus environment_data_provider is a keyword, and should be left out of translation